packages/smooth_app/lib/pages/offline_data.dart

```diff
@@ -0,0 +1,161 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:provider/provider.dart';
+import 'package:smooth_app/database/dao_product.dart';
+import 'package:smooth_app/database/dao_product_list.dart';
+import 'package:smooth_app/database/local_database.dart';
+import 'package:smooth_app/generic_lib/loading_dialog.dart';
+import 'package:smooth_app/pages/preferences/user_preferences_list_tile.dart';
+import 'package:url_launcher/url_launcher.dart';
+
+class OfflineDataScreen extends StatefulWidget {
+  const OfflineDataScreen();
+  @override
+  State<OfflineDataScreen> createState() => _OfflineDataScreenState();
+}
+
+int length = 0;
+double size = 0;
+
+// TODO(ashaman999): update all the applocalizations
+class _OfflineDataScreenState extends State<OfflineDataScreen> {
+  Future<void> _getDetails(DaoProduct daoProduct) async {
+    length = await daoProduct.getLength() ?? 0;
+    size = await daoProduct.getSize();
+    setState(() {});
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final DaoProduct daoProduct = DaoProduct(context.watch<LocalDatabase>());
+    _getDetails(daoProduct);
+    final DaoProductList daoProductList =
+        DaoProductList(context.watch<LocalDatabase>());
+    final MediaQueryData mediaQueryData = MediaQuery.of(context);
+    final bool dark = Theme.of(context).brightness == Brightness.dark;
+    const double titleHeightInExpandedMode = 50;
+    final double backgroundHeight = mediaQueryData.size.height * .20;
+    final Color? foregroundColor = dark ? null : Colors.black;
+    final List<Widget> children = <Widget>[
```

Comment:
Lets put this directly in the SliverList no need for storing here first

Reply:
Done!